dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java

```diff
@@ -740,45 +729,24 @@ protected int createComplementCommandList(Date start, Date end, RunMode runMode,
                         }
                     }
                     logger.info("In parallel mode, current expectedParallelismNumber:{}", createCount);
-
                     // Distribute the number of tasks equally to each command.
                     // The last command with insufficient quantity will be assigned to the remaining tasks.
-                    int itemsPerCommand = (listDateSize / createCount);
-                    int remainingItems = (listDateSize % createCount);
-                    int startDateIndex = 0;
-                    int endDateIndex = 0;
-
-                    for (int i = 1; i <= createCount; i++) {
-                        int extra = (i <= remainingItems) ? 1 : 0;
-                        int singleCommandItems = (itemsPerCommand + extra);
-
-                        if (i == 1) {
-                            endDateIndex += singleCommandItems - 1;
-                        } else {
-                            startDateIndex = endDateIndex + 1;
-                            endDateIndex += singleCommandItems;
+                    for(List<String> stringDate : Lists.partition(listDate,createCount)){
+                        String tempDate = "";
+                        for(String date : stringDate){
+                            tempDate  += date +",";
                         }
-
-                        cmdParam.put(CMDPARAM_COMPLEMENT_DATA_START_DATE, DateUtils.dateToString(listDate.get(startDateIndex)));
-                        cmdParam.put(CMDPARAM_COMPLEMENT_DATA_END_DATE, DateUtils.dateToString(listDate.get(endDateIndex)));
+                        cmdParam.put(CMDPARAM_COMPLEMENT_DATA_SCHEDULE_DATE, tempDate.substring(0,tempDate.length() - 1));
```

Comment:
I think using it's a little confusing using ```substring``` to remove the last comma. Is it more elegant to use ```String.join(",", tmpDateList)``` ?

Reply:
OK，Modified